### DISCUSSION TITLE
List all nodes and iterate recursively?

### DISCUSSION THREAD

**Discussion by Cristo-Conklin** [downforcetech/fluidxml.php#11] [Questions]
Is there something like `$fluid->query('*')`, that would list all nodes?
Other ways to iterate all data without knowing the tag beforehand?

**Comment by daniele-orlando** [comment]
I have not understood what you mean. `$fluid->query('//*')` returns every node in the document, so it's only a matter of what you want to do with them.

``` php
$fluid('//*')->each(function($i, $n) { 
    echo $n->nodeName."\n";
});
```
